### PR TITLE
docs(kfp): choose a workflow executor. Part of kubeflow/pipelines#5718

### PR DESCRIPTION
**Pull request by Bobgy**
Part of https://github.com/kubeflow/pipelines/issues/5718

Preview page: https://deploy-preview-2857--competent-brattain-de2d6d.netlify.app/docs/components/pipelines/installation/choose-executor/

### CODE REVIEW

**Comment by Bobgy** [comment]
/hold
wait for 1.7.0 release to merge the PR

**Review by zijianjoy — COMMENTED** [review]
/lgtm

Awesome documentation! Thank you Yuan for unblocking all KFP users! Maybe we can add the executor options in https://deploy-preview-2857--competent-brattain-de2d6d.netlify.app/docs/components/pipelines/installation/standalone-deployment/#deploying-kubeflow-pipelines, so user can make a decision when they first install KFP.

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+{{% alpha-status feedbacklink="https://github.com/kubeflow/pipelines/issues/6249" %}}
+
+Emissary executor is new. It's first released in Argo Workflows v3.1 (June 2021).
+However, the Kubeflow Pipelines team believe that its architectural, portability
+improvements can make it the default executor that most people should use in the
+future.
+
+Therefore, the team makes a commitment to actively collect feedback and fix bugs
+for the emissary executor, so that we can stablize it faster. So far, Kubeflow
+Pipelines test infra has been running stably with the emissary executor.
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/installation/choose-executor.md:66]: infra -> infrastructure

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+When in doubt, you can always deploy your Kubeflow Pipelines cluster first and
+configure workflow executor after installation using the instructions for
+existing clusters.
+
+##### Migrate pipeline components
+
+`command` must be specified in [Kubeflow Pipelines component specification](https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/).
+This is only for component spec YAML.
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/installation/choose-executor.md:146]: "component spec YAML" -> "the component specification YAML"

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+Example migration:
+
+1. I have a hello world component:
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/installation/choose-executor.md:150]: I'd avoid using "I" in these sentences

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    name: hello-world
+    implementation:
+      container:
+        image: hello-world
+        command: ["/hello"]
+    ```
+
+1. The updated component can be used on emissary executor now.
+
+Note, Kubeflow Pipelines SDK compiler always specifies a command for
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/installation/choose-executor.md:191]: "Note," -> "Note:"

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+[Python function based components](https://www.kubeflow.org/docs/components/pipelines/sdk/python-function-components/).
+Therefore, python function based components will continue to work on emissary
+executor without modifications.
+
+## References
+
+* [Argo Workflow Executors documentation](https://argoproj.github.io/argo-workflows/workflow-executors/)
+* KFP docker executor doesn't support Kubernetes 1.19 or above [kubeflow/pipelines#5714](https://github.com/kubeflow/pipelines/issues/5714)
+* feature request - default to emissary executor [kubeflow/pipelines#5718](https://github.com/kubeflow/pipelines/issues/5718)
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/installation/choose-executor.md:200]: feature -> Feature

File: content/en/docs/components/pipelines/installation/choose-executor.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  * No `privileged` access.
+  * Cannot escape the privileges of the pod's service account.
+* Migration: `command` must be specified in [Kubeflow Pipelines component specification](https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/).
+
+  Note, the same migration requirement is required by [Kubeflow Pipelines v2 compatible mode](https://www.kubeflow.org/docs/components/pipelines/sdk/v2/v2-compatibility/), refer to
+  [known caveats & breaking changes](https://github.com/kubeflow/pipelines/issues/6133).
+
+#### Migrate to Emissary Executor
+
+Prerequisite: emissary executor is only available in Kubeflow Pipelines backend version 1.7+.
```
[the remark this answers is not in the document]
Comment (Bobgy) [reply content/en/docs/components/pipelines/installation/choose-executor.md:81]: Good point! I added a link to the general upgrade entrypoint.

**Comment by Bobgy** [comment]
Thank you @zijianjoy @shuesc1 for the detailed review!
They are super helpful to me.

**Comment by Bobgy** [comment]
/unhold
Because the documentation clearly says it's only supported in 1.7+, so we can merge now.

**Pull request merged**